staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -375,6 +375,39 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 		return readyCheck()
 	}))
 
+	if s.EtcdServersOverrides != nil {
+		if err := s.addOverrideEtcdHealthEndpoint(c); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *EtcdOptions) addOverrideEtcdHealthEndpoint(c *server.Config) error {
+	sc := s.StorageConfig.DeepCopy()
```

Comment:
Why do we have to deep copy the storage config?

Reply:
Updated. Line 392 to user override servers only here.

Reply:
no need to deepcopy... StorageConfig is a struct, and so is StorageConfig.Transport

`storageConfigCopy := s.StorageConfig` is sufficient

Reply:
Updated.